.github/workflows/pytest.yml

```
@@ -0,0 +1,47 @@
+name: Pytest Checks
+
+on: 
+  pull_request:
+    paths:
+      - "tuning/*"
+
+permissions:
+  contents: read
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.1.7
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10.12'
+
+      - name: Install dev dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements-dev.txt
+
+      - name: Set virtual env
+        run: |
+          python -m venv .venv
+          source .venv/bin/activate
+
+      - name: Install tuner dependencies
+        run: |
+          pip install -r tuning/tuner_requirements/requirements.txt
+          python -m pip install iree-compiler
+
+      - name: Run pytest
+        working-directory: tuning
+        run: |
+          python -m pytest test_autotune.py
+          python -m pytest test_tune.py
```

Can we make pytest discover all tests on its own?
```suggestion
          python -m pytest
```